spark/src/test/scala/io/delta/sharing/spark/filters/JsonPredicateSuite.scala

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// scalastyle:off println
+
+package io.delta.sharing.spark.filters
+
+import org.apache.spark.SparkFunSuite
+
+import io.delta.sharing.spark.util.JsonUtils
+
+class JsonPredicateSuite extends SparkFunSuite {
+
+  test("LiteralOp test") {
+    def test_op(op: BaseOp): Unit = {
+      op.validate()
+      val ctx = EvalContext(Map.empty)
+      val (value: String, valueType: String) = op.eval(ctx)
+      assert(value == "2021-04-29")
+      assert(valueType == "date")
+    }
+    val op = LiteralOp(value = "2021-04-29", valueType = "date")
+    test_op(op)
+
+    // Check that we can convert to json and back.
+    val op_from_json = JsonUtils.fromJson[BaseOp](JsonUtils.toJson[BaseOp](op))
+    test_op(op_from_json)
+
+    // Test validation failures.
+    intercept[IllegalArgumentException] {
+      LiteralOp(value = "2021-04-29", valueType = "junk").validate()
+    }
+    intercept[IllegalArgumentException] {
+      LiteralOp(value = null, valueType = "int").validate()
+    }
+  }
```

Comment:
I'll write a lot more unit tests as the review makes progress. Just wanted to get this one out.